components/org.apache.stratos.rest.endpoint/src/main/java/org/apache/stratos/rest/endpoint/api/StratosApiV41.java

```diff
@@ -817,12 +817,18 @@ public Response addApplicationPolicy(
             return Response.created(url).entity(new ResponseMessageBean(ResponseMessageBean.SUCCESS,
                     String.format("Application policy added successfully: [application-policy] %s",
                             applicationPolicy.getId()))).build();
-        } catch (RestAPIException e) {
-            throw e;
         } catch (AutoscalerServiceInvalidApplicationPolicyExceptionException e) {
             return Response.status(Response.Status.BAD_REQUEST).entity(new ResponseMessageBean(
                     ResponseMessageBean.ERROR, "Invalid application policy")).build();
+        } catch(AutoscalerServiceApplicationPolicyAlreadyExistsExceptionException e){
+            return Response.status(Response.Status.CONFLICT).entity(new ResponseMessageBean(
```

Comment:
You can improve the formatting in your diff. There should be a space between ')' and '{'